File: 파이썬/2.검색/1920.py
'''
 * Author : JC
 * https://www.acmicpc.net/problem/1920
'''

def binarySearch(arr, length, target) :
    low = 0
    high = length - 1
    while low <=  high :
        mid = (low + high) // 2
        if target == arr[mid] :
            return 1
        elif target < arr[mid] :
            high = mid - 1
        else :
            low = mid + 1
    return 0

a = int(input())
a1 = list(map(int, input().split()))
b = int(input())
b1 = list(map(int, input().split()))

a1.sort() 

for i in range(b) :
    print(binarySearch(a1, a, b1[i]))

File: C++/27. 이진 검색 알고리즘 - 초급/1920.cpp
/*
 * Author : JC
 * https://www.acmicpc.net/problem/1920
*/

#include <iostream>
#include <algorithm>

using namespace std;

int n, m;

int main() {
        
        cin >> n;
        int arr[n];
        for(int i=0; i<n; i++){
            cin >> arr[i];
        }
        sort(arr, arr+n);
        cin >> m;
        int brr[m];
        for(int i=0; i<m; i++){
            cin >> brr[i];
        }

        for(int i=0; i<m; i++){
            if(binary_search(arr, arr+n, brr[i])>=1){
                cout << 1 << "\n";
            }
            else{
                cout << 0 << "\n";
            }
        }

        return 0;
}